lib/practice_exercises.rb

```diff
@@ -1,13 +1,40 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: 0(n)?
+# Space Complexity: 0(n)?
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string_split = string.upcase.split('')
+  array_letters = string_split.select { |char| /[A-Z]/.match char }
+
+  reverse = []
+  i = array_letters.length
+  while i != 0 do
+    reverse << array_letters[i - 1]
+    i -= 1
+  end
+
+  if reverse == array_letters
+    return true
+  elsif reverse != array_letters
+    return false
+  end
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# # Time Complexity: n^2 ?
+# # Space Complexity: O(1) ?
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍